packages/location-field/src/index.tsx

```diff
@@ -46,8 +48,27 @@ function getOptionId(index: number): string {
   return `${optionIdPrefix}-${index}`;
 }
 
-// FIXME have a better key generator for options
-let optionKey = 0;
+function generateOptionId(feature, indexedOptionLookup) {
+  // Use a default id if the feature does not have an id.
+  let id = "000";
+
+  // User saved locations do not have an id, so use the label.
+  if (feature.displayName) {
+    id = feature.displayName.replace(/\s/g, "");
+  }
+
+  // Some geocoders will return an id. Use that if it exists.
+  if (feature.properties) {
+    id = feature.properties.id || feature.properties.label.replace(/\s/g, "");
```

Comment:
What is the reason for removing blank spaces from the ids? (Add comment)

Reply:
No technical reason, I just rarely see ids that contain whitespace. I can leave it in if it's unneeded!